[AOT Refactor] functional utils

jon-chuang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #114643
* #114564
* #114563
* #114562
* #114561
* #114559
* #114558
* #114557
* #114556
* #114555
* #114554
* #114553
* __->__ #114552
* #114551
* #114550

---

Part _ of https://github.com/pytorch/pytorch/issues/114548

jon-chuang committed:
[AOT Refactor] functional utils

[ghstack-poisoned]

jon-chuang committed:
Update on "[AOT Refactor] functional utils"




[ghstack-poisoned]

jon-chuang committed:
Update on "[AOT Refactor] functional utils"


---

Part _ of https://github.com/pytorch/pytorch/issues/114548


[ghstack-poisoned]

jon-chuang committed:
Update on "[AOT Refactor] functional utils"


---

Part _ of https://github.com/pytorch/pytorch/issues/114548


[ghstack-poisoned]

jon-chuang committed:
Update on "[AOT Refactor] functional utils"


---

Part _ of https://github.com/pytorch/pytorch/issues/114548


[ghstack-poisoned]

jon-chuang committed:
Update on "[AOT Refactor] functional utils"


---

Part _ of https://github.com/pytorch/pytorch/issues/114548


[ghstack-poisoned]

jon-chuang committed:
Update on "[AOT Refactor] functional utils"


---

Part _ of https://github.com/pytorch/pytorch/issues/114548


[ghstack-poisoned]

jon-chuang committed:
Update on "[AOT Refactor] functional utils"


---

Part _ of https://github.com/pytorch/pytorch/issues/114548


[ghstack-poisoned]

jon-chuang committed:
Update on "[AOT Refactor] functional utils"


---

Part _ of https://github.com/pytorch/pytorch/issues/114548


[ghstack-poisoned]

bdhirsh commented on torch/_functorch/_aot_autograd/functional_utils.py:
```diff
@@ -0,0 +1,241 @@
+from torch.utils._python_dispatch import (
+    is_traceable_wrapper_subclass,
+    transform_subclass,
+)
```
Can you add a brief description of which utils are meant for this file? (I think the word `functional` is pretty overloaded)

"This file contains functionalization-related utils used by AOTAutograd"

jon-chuang replied:
Ok, I will add a description to every file

jon-chuang committed:
Update on "[AOT Refactor] functional utils"


---

Part _ of https://github.com/pytorch/pytorch/issues/114548


[ghstack-poisoned]

jon-chuang committed:
Update on "[AOT Refactor] functional utils"


---

Part _ of https://github.com/pytorch/pytorch/issues/114548


[ghstack-poisoned]

jon-chuang commented:
@pytorchbot merge

jon-chuang committed:
Update on "[AOT Refactor] functional utils"


---

Part _ of https://github.com/pytorch/pytorch/issues/114548


[ghstack-poisoned]

jon-chuang commented:
@pytorchbot merge
